### PR TITLE
[dove] Migration to latest feathers dove version

### PR DESCRIPTION
**Pull request by mdartic**
### Summary

This PR address the migration of this package to work in feathers dove.

This is related to #195 too, as the actual package in 4.0.2 is not 100% compatible with dove.

### Other Information

This is still a work in progress.
Compilation is ok, and the `dist/` is working on my test project.
But all tests are not OK. I'm working on it.
I may refer to this PR to ask for help.

### CODE REVIEW

File: examples/js/package.json [hunk 1]
```diff
@@ -37,19 +37,19 @@
...
-    "@feathersjs/authentication": "^4.5.15",
-    "@feathersjs/authentication-local": "^4.5.15",
-    "@feathersjs/authentication-oauth": "^4.5.15",
-    "@feathersjs/configuration": "^4.5.15",
-    "@feathersjs/errors": "^4.5.15",
-    "@feathersjs/express": "^4.5.15",
-    "@feathersjs/feathers": "^4.5.15",
-    "@feathersjs/socketio": "^4.5.15",
-    "@feathersjs/transport-commons": "^4.5.15",
+    "@feathersjs/authentication": "^5.0.0-pre.19",
```
Comment (fratzinger) [inline examples/js/package.json:40]: Please update all dependencies to `pre.29` or even `pre.30`

File: package.json [hunk 2]
```diff
@@ -59,20 +59,21 @@
     "check-updates": "ncu -u && ncu -u --cwd examples/js"
   },
   "dependencies": {
-    "@feathersjs/authentication": "^4.5.15",
-    "@feathersjs/authentication-client": "^4.5.15",
-    "@feathersjs/authentication-local": "^4.5.15",
-    "@feathersjs/errors": "^4.5.15",
+    "@feathersjs/authentication": "^5.0.0-pre.19",
```
Comment (fratzinger) [inline package.json:62]: Please update all dependencies to pre.29 or even pre.30

**Review by fratzinger — COMMENTED** [review]
Thanks for the refactoring! ❤️

**Comment by fratzinger** [comment]
I also opened a new branch: https://github.com/feathersjs-ecosystem/feathers-authentication-management/tree/dove
Could you please make the PR against this branch instead of `master`? Thanks!

File: examples/js/package.json (hunk 1 above)
Comment (mdartic) [reply examples/js/package.json:40]: I have a bug in `pre.30`, there is a new dependency used for the [`typebox` package](https://github.com/feathersjs/feathers/tree/dove/packages/typebox), named `@sinclair/typebox`.
This lib is also used in `schema` package here https://github.com/feathersjs/feathers/blob/dove/packages/schema/src/json-schema.ts#L3.
As it's not added in dependency, it breaks the use of `schema` for TS. (cc @daffl maybe this is for you)

Temporary I add `@sinclair/typebox` in dev dependency to make this compile.

Comment (fratzinger) [reply examples/js/package.json:40]: Yeah, this is obviously a leftover. I created an issue here: https://github.com/feathersjs/feathers/issues/2786

**Comment by fratzinger** [comment]
feathers v5 is not compatible with node12 anymore. So fam does not need to support it as well. So please remove it here:
https://github.com/feathersjs-ecosystem/feathers-authentication-management/blob/master/.github/workflows/node.js.yml#L13
And here: https://github.com/feathersjs-ecosystem/feathers-authentication-management/blob/master/package.json#L29

The minimal supported node version is `>= 14` as can be seen here: https://github.com/feathersjs/feathers/blob/dove/package.json#L23

That's why the CI test for `12.x` fails.

**Comment by mdartic** [comment]
Issues fam+fhc are now understood. They were on my side.
Waiting for `error` feedback.
@fratzinger do you see anything else ?
And do you agree with [my thoughts on `error` package](https://github.com/feathersjs/feathers/issues/2789) ?

**Comment by fratzinger** [comment]
Without any further looking. What about: `const err = new BadRequest(undefined, data)` ?

**Comment by mdartic** [comment]
This "could" work as `BadRequest` constructor signature accept it : 

```ts
export class BadRequest extends FeathersError {
  constructor(message?: ErrorMessage, data?: any) {
    super(message, 'BadRequest', 400, 'bad-request', data)
  }
}
```

But, as `FeathersError` doesn't accept it (`constructor(err: ErrorMessage, name: string, code: number, className: string, _data: any)`), I think either `BadRequest` shouldn't accept a `message?: ErrorMessage` or `FeathersError` should accept a `err?: ErrorMessage`. This seems more consistent ?

Actually, my IDE is not happy with `error` file for this reason.
image

Do you prefer I use your hack ?

File: package.json (hunk 2 above)
Comment (mdartic) [reply package.json]: Updated to pre.31

**Comment by fratzinger** [comment]
Thanks for your work!
Released as `feathers-authentication-management@5.0.0-pre.0` 🎉

**Comment by mdartic** [comment]
Thanks !

**Pull request merged**